src/cromshell/utilities/workflow_id_utils.py

```diff
@@ -0,0 +1,105 @@
+import csv
+import logging
+
+from cromshell.utilities import cromshellconfig, io_utils
+from cromshell.utilities.cromshellconfig import AllWorkflowDatabaseColumns
+
+LOGGER = logging.getLogger(__name__)
+
+
+def resolve_workflow_id(cromshell_input: str, submission_file_path: str) -> str:
+    """
+    Uses input provided by user when running cromshell to identify and return the
+    workflow id from cromshell's submission file
+    :param cromshell_input: User's provided input (digit or alias) in cromshell command
+    :param submission_file_path: path to all.workflow.database.tsv
+    :return: workflow id
+    """
+    if io_utils.is_workflow_id_valid(cromshell_input):
+        return cromshell_input
+    elif cromshell_input.strip("-").isdigit():
+        return obtain_workflow_id_using_digit(
+            relative_id=int(cromshell_input),
+            submission_file_path=submission_file_path,
+        )
+    else:
+        return obtain_workflow_id_using_alias(
+            alias_name=cromshell_input,
+            submission_file_path=submission_file_path,
+        )
+
+
+def obtain_workflow_id_using_digit(relative_id: int, submission_file_path: str) -> str:
+    """
+    Get workflow id from submission file using relative_id.
+    :param relative_id: digit representative of the row of the desired
+    row from submission file.
+    :param submission_file_path: path to all.workflow.database.tsv
+    :return: workflow id
+    """
+    LOGGER.info("Get workflow id from submission file using relative_id.")
+    # If relative id is 0 then use last row
+    if relative_id == 0:
+        LOGGER.error("Relative workflow id must be a non zero integer")
+        raise ValueError("Relative workflow id must be a non zero integer")
+
+    with open(submission_file_path, "r") as csv_file:
+
+        reader = csv.DictReader(csv_file, delimiter="\t")
+        total_rows = len(list(reader))
+        csv_file.seek(0)  # reset file handler after obtaining total lines
+
+        row_index = (
```

Comment:
Possibly jump out the "with" block here since all the required variables are set

Reply:
done